### PR TITLE
Get url from props for prerender-data-provider

### PR DESCRIPTION
**Pull request by mattias-lonn**
**What kind of change does this PR introduce?**
Bugfix

**Did you add tests for your changes?**
No

**Summary**
When example a query is included in the URL path, prerender-data-provider returns null as it doesn't check if the path exists without the query. Now it remaps the URL using given props matches.

**Does this PR introduce a breaking change?**
No

**Other information**
N/A

### CODE REVIEW

**Comment by prateekbh** [comment]
or a test that would demonstrate the same

**Comment by mattias-lonn** [comment]
@ForsakenHarmony: The idea is to ensure that the right json file is fetched by using the props.path instead of props.url. Using getUrlFromProps it will first define the url as props.path and then remap it using props.matches.

For example, if you go to http://localhost:8080/news/news-item-1/?sidebar=login without getUrlFromProps, the json data file path will be `/news/news-item-1/?sidebar=login/preact_prerender_data.json`. The problem is that the query is defined in the url and therefore it will fail to get the json data. This is why I'd remap props.url using props.matches, now the file path will be `/news/news-item-1/preact_prerender_data.json`

Sorry, I do not know how to setup jest for prerender-data-provider folder. @prateekbh, do you have any guidlines to setup a test for prerender-data-provider?
```
./preact-cli/packages/prerender-data-provider/src/hook.test.js
  3:1  error  'describe' is not defined  no-undef
```

File: packages/prerender-data-provider/src/hook.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ function usePrerenderData(props, doAutomaticFetch = true) {
 	});
 	const contextValue = useContext(PrerenderDataContext);
 	checkProps(props);
+	props.url = getUrlFromProps(props);
```
[the remark this answers is not in the document]
Comment (prateekbh) [reply packages/prerender-data-provider/src/hook.js:15]: also there's an HOC

File: packages/prerender-data-provider/src/utils.js [hunk 2]
```diff
@@ -21,4 +21,12 @@ function checkProps(props) {
 	}
 }
 
-export { normalizeUrl, getPrerenderdata, checkProps };
+function getUrlFromProps(props) {
+	let url = props.path;
+	Object.entries(props.matches).forEach(([key, path]) => {
+		url = url.replace(`:${key}`, path);
+	});
+	return url;
```
Comment (prateekbh) [inline packages/prerender-data-provider/src/utils.js:29]: Ideally `normalizeUrl` should also strip query params

**Review by prateekbh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**